yt/fields/derived_field.py

```diff
@@ -589,3 +625,14 @@ def __call__(self, data):
         if getattr(data, "_type_name", None) == "grid":
             return True
         raise NeedsOriginalGrid()
+
+
+def _set_field_validator_base_docstring():
```

Comment:
This would be suited for `FiedlValidator.__init_subclass__`

Reply:
good idea! i'll move it and refactor a bit as needed.

Reply:
now in `__init_subclass__`